### ISSUE TITLE
Combine pachd and worker into single build?

### ISSUE THREAD

**Issue by chainlink** [pachyderm/pachyderm#4764]
## Proposal
Combine pachyderm and worker into single binary and start worker with flag (similar to `doSidecarMode` already in pachd)

This allows us to eliminate the secondary build step for the pachyderm worker binary here:
https://github.com/pachyderm/pachyderm/blob/7f6a123b17550ccdfa8c7a1ac63b94aeedc2571c/Dockerfile#L13

This unlocks faster builds and gets us closer to being able to build a single container for pachd and worker which will simplify our manifests and build pipeline.

Related:
- #4762

**Comment by jdoliner** [comment]
I think this is a good idea, for all the reasons listed here. I'm not sure if there's any reason to do this and not merge the container images. I think the images are basically just the binaries, and some certs. I think the certs are the same, and the binaries will be the same if we do this, so I think the `worker` and `pachd` image will naturally become the same if we do this, and it doesn't seem to make sense to have 2 names for the same container.

**Comment by chainlink** [comment]
Makes sense, I tracked that in https://github.com/pachyderm/pachyderm/issues/4765 because there's a couple extra things we need to do around the init container.

**Comment by jdoliner** [comment]
Yup fine to have it in separate issues, and I'm probably missing something that's going to make it a bit more difficult to combine these than I'm anticipating.

**Comment by RaananHadar** [comment]
Its worth noting that the guys as the Istio project had a similar dilemma recently. They chose to combine the images. They have a pretty good [discussion and pro arguments for it here](https://istio.io/news/releases/1.5.x/announcing-1.5/).
